### DISCUSSION TITLE
Not translated on first render

### DISCUSSION THREAD

**Discussion by barbinbrad** [sergiodxa/remix-i18next#98] [Q&A]
I'm trying to figure out whether this is a bug in my code or "the way it is". I'm guessing I have something wrong in `entry.server.tsx`

I have some code in my component like this: 

```ts
{ t } = useTranslation();

return <h1>{t("displays")}</h1>
```

I noticed that on the first render, the html says `displays` instead of the translated `Displays`. It happens very fast, but here's a video of a video to show the first render:

https://user-images.githubusercontent.com/64510427/192655671-d737adcc-43f5-4f91-9349-91cc1524d6fa.mov

**Comment by sergiodxa** [comment]
This is not how it works, check the entry.server config and ensure your i18next instance has the translation JSON files loaded, how you do that can be either by importing them manually or using an i18next backend file the i18next-fs-backend.

**Reply by barbinbrad** [reply]
Lots of edits. Sorry about that. Here's the final thought:

The server side is fine. If I load the page directly from the server, it renders as it should.

The problem is the client side. When I go from a page that doesn't require namespace X to a page that requires namespace X, where X.json is fetched from the server, there's a problem in the first render. Even if I prefetch, I get the same result. I think I could make a reproduction repo if you'd like.

**Reply by sergiodxa** [reply]
Ah yeah that’s sadly expected, it’s how react-i18next works
